### PR TITLE
feat(Validium): Test `getPubdataPricingMode` getter in `Getters.sol`

### PR DESCRIPTION
**Pull request by toni-calvin**
This PR aims to test the `getPubdataPricingMode` getter in `Getters.sol`. To do so we have called the mentioned method in 2 tests inside the `mailbox_test.spec.ts` to check that once the mode is set, the method returns the correct `pubdataPricingMode`

We have also added the test with  Foundry, the new test suite for this project. 

How to run foundry test: 
- [Install foundry](https://book.getfoundry.sh/getting-started/installation)
- Inside the repository run: `cd l1-contracts/ && yarn test:foundry --match-test test_changePubdataPricingMode`

### CODE REVIEW

File: l1-contracts/contracts/dev-contracts/test/MailboxFacetTest.sol [hunk 1]
```diff
@@ -17,4 +17,8 @@ contract MailboxFacetTest is MailboxFacet {
     function getL2GasPrice(uint256 _l1GasPrice) external view returns (uint256) {
         return _deriveL2GasPrice(_l1GasPrice, REQUIRED_L2_GAS_PRICE_PER_PUBDATA);
     }
+
+    function getPubdataPricingMode() external view returns (PubdataPricingMode) {
```
[the remark this answers is not in the document]
Comment (toni-calvin) [reply l1-contracts/contracts/dev-contracts/test/MailboxFacetTest.sol:21]: You are right thanks! I am going to put this PR in draft until i fix that.

**Comment by ilitteri** [comment]
@StanislavBreadless @vladbochok we're having an issue with the getter being tested. We opened this PR to have a test and make sure this was also happening here because it is already happening to us in the `zksync-era` repo as well.

You can see what I'm saying in the [CI](https://github.com/matter-labs/era-contracts/actions/runs/8082604612/job/22085304121?pr=234) output, but no matter what `PubdataPricingMode` is set, the getter always returns `PubdataPricingMode.Rollup`.

As I said before, this error first came to us in [this PR](https://github.com/matter-labs/zksync-era/pull/1240) whose intention is to cross-check the pubdata pricing mode between `zksync_server` and deployed L1 contracts to ensure that it is the same and panic if not (it is always panicking in the linked PR).

Do you have any idea what could be happening?

**Comment by vladbochok** [comment]
@toni-calvin @ilitteri We are going to switch from hardhat to foundry really soon. Please create new tests in Foundry instead.

Then the issue itself appears because `testContract` is a `Mailbox` facet (implementation contract). `gettersFacet` is a `Getters` facet (implementation contract). They are two separate contracts that share no storage between themself. It is like deploy two ERC20 tokens, change balance in one and expect that another one will change. 

In order to test it properly you must initialise diamond proxy (one contract that delegate calls to Mailbox and Getters while having one storage between both of them).

**Comment by ilitteri** [comment]
> @toni-calvin @ilitteri We are going to switch from hardhat to foundry really soon. Please create new tests in Foundry instead.

Gottcha! We'll move to Foundry then.

> Then the issue itself appears because `testContract` is a `Mailbox` facet (implementation contract). `gettersFacet` is a `Getters` facet (implementation contract). They are two separate contracts that share no storage between themself. It is like deploy two ERC20 tokens, change balance in one and expect that another one will change.
> 
> In order to test it properly you must initialise diamond proxy (one contract that delegate calls to Mailbox and Getters while having one storage between both of them).

This makes sense for the tests to not pass, but we're having the same error on the main repo, where the `DiamondProxy` is initialized correctly. Nevertheless, we're calling the method directly from the Getter contract, I'll change the function call to be directly to the Diamond to see if this fixes the issue.

**Comment by toni-calvin** [comment]
> @toni-calvin @ilitteri We are going to switch from hardhat to foundry really soon. Please create new tests in Foundry instead.

We have added the test in foundry and is working as expected
 
> In order to test it properly you must initialise diamond proxy (one contract that delegate calls to Mailbox and Getters while having one storage between both of them).

Thank you for your answer! We updated the test as you suggested it and it worked as expected. 

This PR is ready for review @vladbochok

**Pull request closed without merging**